pkg/kubelet/dockertools/labels_test.go

```diff
@@ -0,0 +1,67 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package dockertools
+
+import (
+	"strconv"
+	"testing"
+
+	"k8s.io/kubernetes/pkg/api"
+	kubecontainer "k8s.io/kubernetes/pkg/kubelet/container"
+)
+
+func TestLabels(t *testing.T) {
+	restartCount := 5
+	container := api.Container{
+		Name: "test_container",
+		TerminationMessagePath: "/tmp",
+	}
+	pod := api.Pod{
+		ObjectMeta: api.ObjectMeta{
+			Name:      "test_pod",
+			Namespace: "test_pod_namespace",
+			UID:       "test_pod_uid",
+		},
+	}
+	labels := newLabels(&container, &pod, restartCount)
+	containerInfo, err := getContainerInfoFromLabel(labels)
+	if err != nil {
+		t.Errorf("Unexpected error when getContainerInfoFromLabel: %v", err)
+	}
+	if containerInfo.PodName != pod.Name {
+		t.Errorf("Pod name: expected %s, got %s", pod.Name, containerInfo.PodName)
+	}
+	if containerInfo.PodNamespace != pod.Namespace {
+		t.Errorf("Pod namespace: expected %s, got %s", pod.Namespace, containerInfo.PodNamespace)
+	}
+	if containerInfo.PodUID != pod.UID {
+		t.Errorf("Pod UID: expected %s, got %s", pod.UID, containerInfo.PodUID)
+	}
+	if containerInfo.Name != container.Name {
+		t.Errorf("Container name: expected %s, got %s", container.Name, containerInfo.Name)
+	}
+	containerHash := strconv.FormatUint(kubecontainer.HashContainer(&container), 16)
+	if containerInfo.Hash != containerHash {
+		t.Errorf("Container hash: expected %s, got %s", containerHash, containerInfo.Hash)
+	}
+	if containerInfo.RestartCount != restartCount {
+		t.Errorf("Container restartCount: expected %d, got %d", restartCount, containerInfo.RestartCount)
+	}
+	if containerInfo.TerminationMessagePath != container.TerminationMessagePath {
+		t.Errorf("Container terminationMessagePath: expected %s, got %s", container.TerminationMessagePath, containerInfo.TerminationMessagePath)
```

Comment:
Done. In fact I remove all these in new commit. However, I learned that I should use %q for printing string, thanks :)